### DISCUSSION TITLE
AC with ecdsa p521  sha512withEcdsa  and  pkcs10  with  RSA.  can?  Sign PDF

### DISCUSSION THREAD

**Discussion by jpertuz-19** [Keyfactor/ejbca-ce#618] [General]
Greetings. 

We have an AC that was generated with ecdsa p521.

defaultKey - ECDSA P-521
encryptKey - ECDSA P-521
signKey - ECDSA P-521
testKey - ECDSA P-521

Signing Algorithm sha512WithECDSA

My Perfil Certificate  for the Entity:
accept ECDSA andRSA ,   with Signing Algorithm sha512WithECDSA. 

We are issuing certificates for signing PDF documents.  
1) user generates private key with rsa, but with AC RSA through JS libraries (development) we can sign PDF.

But with ac ECDSA and ecdsa private key we have problems being able to read the certificate and sign.  Could it be the generation of the private key? 

Add. Can I use a CA with ECDSA and generate certificates that were requested (pkcs10) with rsa private key?

I have questions about signing Digital documents (PDF) and elliptic curve support.

**Comment by primetomas** [comment]
You can absolutely issue a certificate with a RSA subject public Key, the certificate signed by your EC CA using ECDSA. No problem to mix and match.

Related to your PDF signing that should be something with that implementation. EC certificates issued by EJBCA is widely deployed and used.
